Sources/StreamChat/StateLayer/ChannelListState+Observer.swift

```diff
@@ -31,6 +50,39 @@ extension ChannelListState {
         }
         
         func start(with handlers: Handlers) {
+            /// When we receive events, we need to check if a channel should be added or removed from
+            /// the current query depending on the following events:
+            /// - Channel created: We analyse if the channel should be added to the current query.
+            /// - New message sent: This means the channel will reorder and appear on first position,
+            ///   so we also analyse if it should be added to the current query.
+            /// - Channel is updated: We only check if we should remove it from the current query.
+            ///   We don't try to add it to the current query to not mess with pagination.
+            let nc = eventNotificationCenter
+            eventObservers = [
+                EventObserver(notificationCenter: nc, transform: { $0 as? NotificationAddedToChannelEvent }) { [weak self] event in
+                    try await self?.linkChannelIfNeeded(event.channel)
+                },
+                EventObserver(notificationCenter: nc, transform: { $0 as? MessageNewEvent }) { [weak self] event in
+                    try await self?.linkChannelIfNeeded(event.channel)
+                },
+                EventObserver(notificationCenter: nc, transform: { $0 as? NotificationMessageNewEvent }) { [weak self] event in
+                    try await self?.linkChannelIfNeeded(event.channel)
+                },
+                EventObserver(notificationCenter: nc, transform: { $0 as? ChannelUpdatedEvent }) { [weak self] event in
+                    try await self?.unlinkChannelIfNeeded(event.channel)
+                },
+                EventObserver(notificationCenter: nc, transform: { $0 as? ChannelVisibleEvent }) { [weak self] event in
+                    guard let self else { return }
+                    let channel = try await self.database.backgroundRead { context in
+                        guard let dto = ChannelDTO.load(cid: event.cid, context: context) else {
+                            throw ClientError.ChannelDoesNotExist(cid: event.cid)
+                        }
+                        return try dto.asModel()
+                    }
+                    try await self.linkChannelIfNeeded(channel)
+                }
+            ]
```

Comment:
Similar to what we have in [ChannelListController](https://github.com/GetStream/stream-chat-swift/blob/develop/Sources/StreamChat/Controllers/ChannelListController/ChannelListController.swift#L266). Only change is what @martinmitrevski suggested in the previous PR which is using EventObservers.